tests/wallet/rpc/test_wallet_rpc.py

```diff
@@ -1125,6 +1125,15 @@ async def test_select_coins_rpc(wallet_rpc_environment: WalletRpcTestEnvironment
         assert coin != coin_300[0]
 
 
+@pytest.mark.asyncio
+async def test_reset_wallet_sync(wallet_rpc_environment: WalletRpcTestEnvironment):
+    env: WalletRpcTestEnvironment = wallet_rpc_environment
+    full_node_api: FullNodeSimulator = env.full_node.api
+    client: WalletRpcClient = env.wallet_1.rpc_client
+    await generate_funds(full_node_api, env.wallet_1)
+    await client.reset_wallet_sync()
```

Comment:
there's no test here. If you make `reset_wallet_sync()` an empty function, this test will still pass.

It needs to ensure both that the intended state was achieved and also that it's safe to do while performing any other wallet operation. This test isn't covering any interesting edge cases that (in my mind) are quite likely to cause problems.

Reply:
TBH, I don't know how to simulate such complicated cases in unit tests. If you know how to do it let me know.

Reply:
this functionality is likely to break in the future unless we have an automated test for it.
maybe you could have the test check the wallet log to ensure it resynced